From e3945a012814b6bd91d065e6fa2f75c6a0a7fc98 Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tomas@openssl.org>
Date: Thu, 11 Apr 2024 11:34:57 +0200
Subject: [PATCH 123/228] test/prov_config_test.c: Cleanup and fix potential
 leaks

Fixes #24106

Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
Reviewed-by: Neil Horman <nhorman@openssl.org>
Reviewed-by: Paul Dale <ppzgs1@gmail.com>
(Merged from https://github.com/openssl/openssl/pull/24107)

(cherry picked from commit 140540189c67ba94188165b1144fdfb5b248bc02)
---
 test/prov_config_test.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/test/prov_config_test.c b/test/prov_config_test.c
index f93d8d62be..2fd913dead 100644
--- a/test/prov_config_test.c
+++ b/test/prov_config_test.c
@@ -26,15 +26,13 @@ static int test_double_config(void)
     int testresult = 0;
     EVP_MD *sha256 = NULL;
 
-    if (!TEST_ptr(configfile))
-        return 0;
     if (!TEST_ptr(ctx))
         return 0;
 
     if (!TEST_true(OSSL_LIB_CTX_load_config(ctx, configfile)))
-        return 0;
+        goto err;
     if (!TEST_true(OSSL_LIB_CTX_load_config(ctx, configfile)))
-        return 0;
+        goto err;
 
     /* Check we can actually fetch something */
     sha256 = EVP_MD_fetch(ctx, "SHA2-256", NULL);
@@ -54,9 +52,6 @@ static int test_recursive_config(void)
     int testresult = 0;
     unsigned long err;
 
-    if (!TEST_ptr(recurseconfigfile))
-        goto err;
-
     if (!TEST_ptr(ctx))
         goto err;
 
-- 
2.39.5 (Apple Git-154)

